src/Hpack/Render.hs

```diff
@@ -252,6 +252,7 @@ renderSection renderSectionData extraFieldsStart extraFieldsEnd Section{..} = ad
   ++ maybe [] (return . renderBuildable) sectionBuildable
   ++ map (renderConditional renderSectionData) sectionConditionals
   ++ extraFieldsEnd
+  ++ maybe [] (return . renderDefaultLanguage) sectionDefaultLanguage
```

Comment:
Putting this at the end will prevent you from overriding it in a conditional.  Conditionals have to go last.